src/Core/src/Platform/iOS/MauiUISceneDelegate.cs

```diff
@@ -32,6 +32,15 @@ public virtual void WillConnect(UIScene scene, UISceneSession session, UISceneCo
 		public virtual void DidDisconnect(UIScene scene)
 		{
 			MauiUIApplicationDelegate.Current?.Services?.InvokeLifecycleEvents<iOSLifecycle.SceneDidDisconnect>(del => del(scene));
+
+			if (Window is not null && Window.IsKeyWindow)
+			{
+				// manually resign the key window and rebuild the menu
+				Window.ResignKeyWindow();
+				UIMenuSystem
```

Comment:
I agree, but I found a limitation on MacCatalyst when research for this problem. There aren't any API we can pull to get the focused window. It may sound crazy but iPad actually doesn't need this since only one window will be visible at a time which is the latest window created (also key window). The only API that can track focused windows I found is from AppKit (is there a way we can access AppKit?). That is why my fix only can play around with keywindow. Let me know if you have better idea.

Reply:
I tried this and it seems to work on catalyst
https://stackoverflow.com/a/75440759

I think you can check if the activeappearance on the UIWindows's traitcollection is active.

Reply:
Previously I tried this solution and it doesn't return the focused window but activated scene only. But I was testing many different methods that time and I might be wrong, let me try again and get back to here later.

Reply:
The active appearance work with some workaround, see my latest changes, now iOS14+ can rebuild menu whenever a window get focused.